File: src/Test/PrimesFFICheatcode.fs
module Test.PrimesFFICheatcode
let isPrime n =
    if n < 2 then false
    else
        let mutable p = true
        let mutable i = 2
        while p && i * i <= n do
            if n % i = 0 then p <- false
            i <- i + 1
        p
let runPrimesFFICheatcode (n: obj) =
    let n' = unbox<int> n
    let mutable sum = 0
    for i in 2 .. n' do
        if isPrime i then sum <- sum + i
    sum :> obj

File: scratch/record-scalarization-integration-20260916/step1/src/Test/PrimesFFICheatcode.cc
#include "purescript.h"
#include <vector>
FOREIGN_BEGIN( Test_PrimesFFICheatcode )
exports["runPrimesFFICheatcode"] = [](const boxed& in) -> boxed {
    const int n = unbox<int>(in);
    if (n < 2) return 0;
    std::vector<bool> is_prime(n + 1, true);
    int sum = 0;
    for (int p = 2; p <= n; p++) {
        if (is_prime[p]) {
            sum += p;
            for (std::size_t multiple = std::size_t(p) * p;
                 multiple < is_prime.size(); multiple += p) {
                is_prime[multiple] = false;
            }
        }
    }
    return sum;
};
FOREIGN_END
